questionary/prompts/path.py

```diff
@@ -0,0 +1,129 @@
+import os
+from typing import (
+    Any,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+)
+
+from prompt_toolkit.completion import CompleteEvent, Completion, PathCompleter
+from prompt_toolkit.document import Document
+from prompt_toolkit.key_binding import KeyBindings
+from prompt_toolkit.key_binding.key_processor import KeyPressEvent
+from prompt_toolkit.keys import Keys
+from prompt_toolkit.lexers import SimpleLexer
+from prompt_toolkit.shortcuts.prompt import CompleteStyle, PromptSession
+from prompt_toolkit.styles import Style, merge_styles
+
+from questionary.constants import DEFAULT_QUESTION_PREFIX, DEFAULT_STYLE
+from questionary.prompts.common import build_validator
+from questionary.question import Question
+
+
+class GreatUXPathCompleter(PathCompleter):
+    def get_completions(
+        self, document: Document, complete_event: CompleteEvent
+    ) -> Iterable[Completion]:
+        """Get completions.
+
+        Wraps `PathCompleter`. Makes sure completions for directories end with
+        a path separator. Also make sure the right path separator is used."""
+        completions = super(GreatUXPathCompleter, self).get_completions(
+            document, complete_event
+        )
+
+        for completion in completions:
+            # check if the display value ends with a path separator.
+            # first check if display is properly set
+            styled_display = completion.display[0]
+            # styled display is a formatted text (a tuple of the text and its style)
+            # second tuple entry is the text
+            if styled_display[1][-1] == "/":
+                # replace separator with the OS specific one
+                display_text = styled_display[1][:-1] + os.path.sep
+                # update the styledd display with the modified text
+                completion.display[0] = (styled_display[0], display_text)
+                # append the separator to the text as well - unclear why the normal
+                # path completer omits it from the text. this improves UX for the
```

Comment:
I believe the reason this is not appended automatically is that the slash is what triggers the initial completion.  If you start typing something, hit tab to complete it and then the slash is appended automatically for you, the completions for the next portion of the path won't appear - you'll have to hit backspace and re-insert the slash.  Hitting <tab> brings up the completion for the already-completed previous portion of the path.

Reply:
@layday `TAB` does bring up completion suggestions, at least for me?

the workflow I aimed for was: type, hit tab to open and select completion, hit enter to confirm completion, continue typing OR hit tab to complete the next segment

Reply:
It brings up the completion suggestions for the already-completed path:

![completion](https://user-images.githubusercontent.com/31134424/97981700-5a99bc00-1ddb-11eb-9cbd-8e22dd4a9ed0.gif)

Notice how I had to insert a second `/` to bring up the suggestions for the next portion of the path.  Inserting a slash marks the end of the previous completion.

Reply:
you can also confirm the auto completion with enter instead of your second `/`

Reply:
Ah, I just assumed that enter would submit the answer regardless like prompt-toolkit would do.  It makes sense that way, thanks.

Reply:
I've slightly changed the implementation, do you mind checking if it is more intuitive for you now?

Reply:
Sorry, I forgot to reply. It works great!